src/main/js/helpers/eventline.js

```diff
@@ -0,0 +1,145 @@
+ * When a new content is loaded, the height of the graph changes
+ * since the height is consumer provided value. We would need to
+ * translate the straight line based on the new height.
+ *
+ * @private
+ * @param {object} scale - d3 scale taking into account the input parameters
+ * @param {object} config - config object derived from input JSON
+ * @param {Array} canvasSVG - d3 object of canvas svg
+ * @param {Function} yAxisPositionHandler - call back to get y-axis position.Functions differ based on Constructs used
+ * @returns {undefined} - returns nothing
+ */
+const translateEventline = (scale, config, canvasSVG, yAxisPositionHandler) => {
+    if (utils.isEmpty(config.eventline)) {
+        return;
+    }
+    const eventlineGroup = canvasSVG
+        .selectAll(`.${styles.eventlineGroup}`)
+        .transition()
+        .call(constants.d3Transition(config.settingsDictionary.transition))
+        .attr(
+            "transform",
+            `translate(${getXAxisXPosition(config)},${yAxisPositionHandler(
+                config
+            )})`
+        );
+    eventlineGroup
+        .selectAll(`.${styles.eventlinePoint}`)
+        .select("g")
+        .attr("transform", function(d) {
+            return `translate(${scale.x(
+                utils.parseDateTime(d.value)
+            )}, 0) scale(${getTransformScale(this)})`;
+        });
+    eventlineGroup
+        .selectAll(`.${styles.eventline}`)
+        .attr("x1", (val) => scale.x(utils.parseDateTime(val.value)))
+        .attr("y1", 0)
+        .attr("x2", (val) => scale.x(utils.parseDateTime(val.value)))
+        .attr("y2", getYAxisHeight(config));
+};
+/**
+ * Creates a eventline for graph. We are not adding
+ * x1, x2, y1, y2 co-ordinates for the straight line since they will be adjusted when
+ * content is loaded.
+ *
+ * @private
+ * @param {object} scale - d3 scale taking into account the input parameters
+ * @param {object} config - config object derived from input JSON
+ * @param {Array} canvasSVG - d3 object of canvas svg
+ * @returns {undefined} - returns nothing
+ */
+const createEventline = (scale, config, canvasSVG) => {
+    const eventlineContent = canvasSVG
+        .append("g")
+        .classed(styles.eventlineContent, true)
+        .attr("clip-path", `url(#${config.clipPathId})`);
+
+    config.eventline.forEach((eventline, index) => {
+        const eventlineGroup = eventlineContent
+            .append("g")
+            .classed(styles.eventlineGroup, true)
+            .datum(eventline)
+            .attr("style", (val) => `fill: ${val.color}; stroke: ${val.color};`)
+            .attr("aria-selected", false);
+
+        eventlineGroup
+            .append("line")
+            .classed(styles.eventline, true)
+            .attr(
+                "style",
+                (d) =>
+                    `fill: ${d.color}; stroke: ${d.color}; stroke-dasharray: ${
+                        d.style !== undefined &&
```

Comment:
I see that it would be a solid line, if nothing is provided. I think we should have a regular dotted line instead, since this needs to be distinct from a dateline. 
How does `2,2` sound?

Reply:
Since dateline is slight opac and if we don't provide any style the eventline will be solid and opacity will be 1, so if we need a more solid line we can use eventline for it. If it really needs to have a default 2,2 value then I will change it to the same.